service-base/src/main/java/org/eclipse/hono/service/registration/BaseRegistrationService.java

```diff
@@ -212,16 +218,21 @@ public void assertRegistration(
         Objects.requireNonNull(deviceId);
         Objects.requireNonNull(resultHandler);
 
+        final Span currentSpan = buildSpan(spanContext, tenantId, deviceId, null);
```

Comment:
hmm, given that this method is supposed to be overridden by users, wouldn't it make sense to create a Span in *processAssertRequest* already so that we have at least *some* kind of tracing in any case (i.e. regardless of whether this method is overridden or not)?

Reply:
As the `assertRegistration` method isn't necessarily overridden by users, I don't see this as strictly necessary.
For the case that the default `assertRegistration` method is used, an extra span in `processAssertRequest` doesn't provide much extra info. And creating such a Span in `processAssertRequest` (while making sure it is finished when the returned future is finised) would make the code more complex.
I would say, if a user wants to implement a custom `assertRegistration` method, the user should also implement tracing there if tracing is relevant for the user. I've make the method to create a span protected, so that adding tracing support in an overridden method should be straightforward.

Reply:
FMPOV it is the other way around. Because the *assertRegistration* **may** be overridden, we should start (and finish) a span in *processAssertRequest* and pass in that span into *assertRegistration*. A subclass then may (or may not) log to that span or create child spans of it as it sees fit. But in any case we have some minimal trace of the *assertRegistration* method being executed (and of its outcome).

Reply:
The extra, kind of superfluous span being created in the (probably common) case that `assertRegistration` is _not_ overridden doesn't look that good, FMPOV.
I've added the span nonetheless and it at least has the benefit of capturing the case of a missing tenant/device id.